src/components/TodoList.tsx

```diff
@@ -0,0 +1,243 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+
+interface Props {
+  todos: Todo[];
+  deleteTodo: (id: number) => Promise<void>;
+  deleteDisabled: number[];
+  tempTodo: Todo | null;
+  deleteCompleted: () => Promise<void>;
+  clearDisabled: boolean;
+}
+
+export const TodoList: React.FC<Props> = ({
+  todos: todosData,
+  deleteTodo,
+  deleteDisabled,
+  tempTodo,
+  deleteCompleted,
+  clearDisabled,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>(todosData);
+  const [filterBy, setFilterBy] = useState<string>('all');
```

Comment:
Create enum for all statuses
```suggestion
  const [filterBy, setFilterBy] = useState<StatusType>(StatusType.All);
```